secboot/secboot_tpm.go

```diff
@@ -216,10 +217,22 @@ func UnlockVolumeIfEncrypted(name string, lockKeysOnFinish bool) (string, error)
 			}
 		}()
 
-		ok, encdev := isDeviceEncrypted(name)
-		if !ok {
+		// find the encrypted device using the disk we were provided - note that
+		// we do not specify IsDecryptedDevice in opts because here we are
+		// looking for the encrypted device to unlock, later on in the boot
+		// process we will look for the decrypted device to ensure it matches
+		// what we expected
+		partUUID, err := disk.FindMatchingPartitionUUID(name + "-enc")
+		var errNotFound disks.FilesystemLabelNotFoundError
+		if xerrors.As(err, &errNotFound) {
+			// didn't find the encrypted label, so return nil to try the
+			// decrypted label again
 			return nil
```

Comment:
mmh, instead shouldn't we maybe simply return as well whether we decrpyted or not so that the caller doesn't need to check/guess again?

Reply:
yes that ended up being fine to do